internal/components/monitor/helpers.go

```diff
@@ -0,0 +1,45 @@
+package monitor
+
+import (
+	"fmt"
+	"slices"
+)
+
+// ValidateMonitoringOperation checks if the monitoring operation is supported
+func ValidateMonitoringOperation(operation string) bool {
+	supportedOps := []string{
+		string(OpMetrics), string(OpResourceHealth), string(OpAppInsights),
+		string(OpDiagnostics), string(OpControlPlaneLogs),
+	}
+	return slices.Contains(supportedOps, operation)
+}
+
+// GetSupportedMonitoringOperations returns all supported monitoring operations
+func GetSupportedMonitoringOperations() []string {
+	return []string{
+		string(OpMetrics), string(OpResourceHealth), string(OpAppInsights),
+		string(OpDiagnostics), string(OpControlPlaneLogs),
+	}
+}
```

Comment:
This function creates a new slice on every call. Instead, it should return the same supportedOperations slice used by ValidateMonitoringOperation() to eliminate code duplication and improve efficiency.